File: src/go/problems/total_money_test.go
// https://leetcode.com/problems/calculate-money-in-leetcode-bank

package problems

import (
	"testing"

	"github.com/dobermanch/leetcode/core"
)

type TotalMoney struct{}

func TestTotalMoney(t *testing.T) {
	gen := core.TestSuite[TotalMoney]{}
	gen.Add(func(tc *core.TestCase) {
		tc.Param(4).Result(10)
	}).Add(func(tc *core.TestCase) {
		tc.Param(10).Result(37)
	}).Add(func(tc *core.TestCase) {
		tc.Param(20).Result(96)
	}).Run(t)
}

func (TotalMoney) Solution(n int) int {
	week := 0
	result := 0

	for i := 0; i < n; i++ {
		day := i % 7
		if day == 0 {
			week++
		}

		result += week + day
	}

	return result
}

File: src/python/problems/total_money_test.py
# https://leetcode.com/problems/calculate-money-in-leetcode-bank

from core.problem_base import *

class TotalMoney(ProblemBase):
    def Solution(self, n: int) -> int:
        week = 0
        result = 0
        for i in range(n):
            day = i % 7
            if day == 0:
                week += 1
            
            result += week + day

        return result

if __name__ == '__main__':
    TestGen(TotalMoney) \
        .Add(lambda tc: tc.Param(4).Result(10)) \
        .Add(lambda tc: tc.Param(10).Result(37)) \
        .Add(lambda tc: tc.Param(20).Result(96)) \
        .Run()
